content/documentation/runtimes-monitoring/index.adoc

```diff
@@ -0,0 +1,228 @@
+
+When installing Kiali, a new CRD is installed in the system: _monitoringdashboard.monitoring.kiali.io_. It declares the resource kind _MonitoringDashboard_. Here's how this resource looks like:
+
+```yaml
+apiVersion: "monitoring.kiali.io/v1alpha1"
+kind: MonitoringDashboard
+metadata:
+  name: vertx-server
+spec:
+  runtime: Vert.x
+  title: Vert.x Server Metrics
+  charts:
+    - name: "Server response time"
+      unit: "s"
+      spans: 6
+      metricName: "vertx_http_server_responseTime_seconds"
+      dataType: "histogram"
+      aggregations:
+        - label: "path"
+          displayName: "Path"
+        - label: "method"
+          displayName: "Method"
+    - name: "Server active connections"
+      unit: ""
+      spans: 6
+      metricName: "vertx_http_server_connections"
+      dataType: "raw"
+```
+
+The *name* field (from metadata) corresponds to what you will have to set in pods annotation `kiali.io/runtimes`.
+
+Spec fields definitions are:
+
+* *runtime*: name of the related runtime. It will be displayed on the corresponding Workload Details page. Can be omitted to not display anything.
+* *title*: dashboard title, displayed as a tab in App or Workloads Details
+* *charts*: list of charts in the dashboard
+** *name*: name of the chart
+** *unit*: unit for Y-axis. This unit may be converted using metric SI system, for instance "s" can be shown as "ms" or "µs" as well.
+** *spans*: number of "spans" taken by the chart, from 1 to 12, using link:https://www.w3schools.com/bootstrap4/bootstrap_grid_system.asp[bootstrap convention]
+** *metricName*: the metric name in Prometheus
+** *dataType*: type of data to be displayed in the chart. Can be one of _raw_, _rate_ or _histogram_. Raw data will be queried without transformation. Rate data will be queried using link:https://prometheus.io/docs/prometheus/latest/querying/functions/#rate[_promQL rate() function_]. And histogram with link:https://prometheus.io/docs/prometheus/latest/querying/functions/#histogram_quantile[_histogram_quantile() function_].
+** *aggregator*: defines how the time-series are aggregated when several of them are returned for a given metric and labels set. For example, if a Deployment creates a ReplicaSet of several Pods, you will have at least one time-series per Pod. Since Kiali shows the dashboards at the Workload (ReplicaSet) level or at the App level, they will have to be aggregated. This field can be used to fix the aggregator, with values such as _sum_ or _avg_ (full list available link:https://prometheus.io/docs/prometheus/latest/querying/operators/#aggregation-operators[in Prometheus documentation]). However, you can omit it: in that case, the aggregator will default to _sum_ and can be changed from the dashboard UI.
+** *aggregations*: list of labels eligible for aggregations / groupings (they will be displayed in Kiali through a dropdown list)
```

Comment:
while writing this doc, I feel the word "aggregations" is not really accurate here, I wonder if I should rename to something like "grouping", as it allows to group series by selected labels.
Renaming here would mean change the data model in Kiali.

Reply:
minor: "when several of them are returned for a given metric and labels set" -> "when several are returned for a given metric and label set"
minor: "However, you can omit it: in that case, the" -> "However, if omitted the"

As for "aggregations", it's not perfect because an aggregation is typically a result.  But alternatives aren't obvious.  Prometheus uses "by", which is sort or short for "groupBy".   Can you select only one label from the list, or several?  If one then "groupBy" or "groupings" could work.  If you can select multiple then "groupBy", "groupByLabels" or "groupingLabels" could work.    I'm OK if you stick with "aggregations" although I can understand if you want to change to one of the "group" variants.